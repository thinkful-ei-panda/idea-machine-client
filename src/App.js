import React from 'react';
import {Route, Switch} from 'react-router-dom';

import Header from './components/Header/Header'
import SearchBarPage from './components/SearchBarPage/SearchBarPage'
import LoginPage from './routes/LoginPage'
import AddIdeaPage from './routes/AddIdeaPage';

import PublicOnlyRoute from './components/Utils/PublicOnlyRoute';
import PrivateRoute from './components/Utils/PrivateRoute';
import TokenService from './services/token-service';
import MyIdeasPage from './routes/MyIdeasPage';
import TrackedIdeasPage from './routes/TrackedIdeasPage';
import RegistrationPage from './routes/RegistrationPage';
import EditIdeaPage from './routes/EditIdeaForm';



class App extends React.Component {
  state = {
    loggedInToggle:false,
    editIdeaValues: {
      title:null,
      content:null
    }
  }

  handleLogIn = (history) => {
    this.setState({loggedInToggle:true})

    history.push('/')
  }

  handleLogout = () => {
    TokenService.clearAuthToken()
    this.setState({loggedInToggle:false})
  }

  handleEditClick = (e,history,title,content) => {
    this.setState({editIdeaValues: {
      title,
      content,
      id: e.target.closest('li').id,
    }},() => history.push('/edit-idea-page'))
  }

  render(){
    return (
      <div className="App">
        <Header loggedInToggle={this.state.loggedInToggle}  handleLogout={this.handleLogout} />
        <main>
          <Switch>
            <Route
            exact
            path='/'
            component={SearchBarPage} />
  
            <PublicOnlyRoute
            exact
            path='/login'
            component={props => <LoginPage {...props} handleLogIn={this.handleLogIn}/>}
            />

            <PublicOnlyRoute
            exact
            path='/register'
            handleLogIn={this.handleLogIn}
            component={props => <RegistrationPage {...props} handleLogIn={this.handleLogIn}/>}             
            />

            <PrivateRoute
            exact
            path='/my-ideas'            
            component = {props => <MyIdeasPage {...props} handleEditClick={this.handleEditClick}/> }
            />
            
            <PrivateRoute
            exact
            path='/add-idea'
            component={props => <AddIdeaPage {...props} />}          
             />
             
            <PrivateRoute
            exact
            path='/edit-idea-page'
            component={props => <EditIdeaPage {...props} editIdeaValues={this.state.editIdeaValues}/>}            
             />
  
            <PrivateRoute
            exact
            path='/tracked-ideas'
            component={TrackedIdeasPage} />
  
          </Switch>
        </main>
      </div>
    );
  } 
}

export default App;
