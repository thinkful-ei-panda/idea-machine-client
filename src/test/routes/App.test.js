import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from '../../App';
describe(`App Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <App />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <App />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})