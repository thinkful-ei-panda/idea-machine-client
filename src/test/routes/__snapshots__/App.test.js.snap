// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="App"
>
  <header
    className="Header"
  >
    <h1>
      <a
        href="/home"
        onClick={[Function]}
      >
        ID8
      </a>
    </h1>
    <div
      className="NavContainer"
    >
      <nav
        className="Nav"
      >
        <a
          aria-current={null}
          href="/home"
          onClick={[Function]}
        >
          <div>
            Home
          </div>
        </a>
        <a
          aria-current={null}
          href="/login"
          onClick={[Function]}
        >
          <div>
            Login
          </div>
        </a>
        <a
          aria-current={null}
          href="/register"
          onClick={[Function]}
        >
          <div>
            Register
          </div>
        </a>
      </nav>
    </div>
  </header>
  <main>
    <div
      className="Landing-Page"
    >
      <section
        className="instructions"
      >
        Save ideas and get inspired
      </section>
      <button
        onClick={[Function]}
      >
        Inspiration
      </button>
    </div>
  </main>
</div>
`;
